rfcs/convergence/deprecate-keyboard-key-package.md

```diff
@@ -0,0 +1,82 @@
+
+| Everyone else | IE11     |
+| ------------- | -------- |
+| ArrowDown     | Down     |
+| Escape        | Esc      |
+| ' '           | Spacebar |
+
+## Detailed Design or Proposal
+
+### Remove dual v8/v9 dependency on `@fluentui/keyboard-key`
+
+Since there are v9 components now being used in HVCs and products, this might be difficult since this could result in a breaking change in terms of dependencies.
+
+#### Option 1: Keep using keyboard-key
+
+We can copy the contents of keyboard-key internally into the v8 `react` folder. We have already done this previously with v7. This lets us modify keyboard-key freely without breaking v8.
+
+However this will be messy as products migrate to v9. For example teams has a specific yarn resolution to a version of `@fluentui/keyboard-key`
+
+#### Option 2: Use existing package (i.e. react-utilities)
+
+Since we expect any new form key(code)s to be easily tree shakeable we can simply use an existing package like `@fluentui/react-utilities` to store the next tree-shakeable iteration of these key(code)s.
+
+If following semver, this would not result in a major bump of the package and should be updated without too many problems by partners.
+
+> NOTE: we do not currently use semver, so we might not have this advantage while in alpha
+
+#### Option 3: Create new package
+
+We could create a new `@fluentui/react-keyboard-key` (naming flexible) package to store these constants. The one advantage this would have over the previous option is more readable imports:
+
+```typescript
+import { ArrowDown, Escape } from '@fuentui/react-keyboard-key';
+```
+
+instead of
+
+```typescript
+import { ArrowDown, useIsomorphicLayoutEffect, Escape, useEventCallback } from '@fuentui/react-utilities';
+```
+
+> The below statement should be verified
+
+There should be no major problem with dependencies since older packages will still work with the old keyboard-key package. We generally do not share any logic with keys in our shared contexts or cross components.
+
+### Use e.key
+
+Since we no longer target support for IE11, the usage of `e.key` is simplified and no longer needs to have special handling for IE11. We can modify an existing package or create a new one that simply stores all of the keys as constants which are easily tree shaken.
+
+## Pros and Cons
+
+### Pros
+
+- Continues our original objective of separating v8 and v9 dependencies
+- Stop using deprecated properties
+- Result in a ~2kb minified size decrease for every package that uses keyboard-key (all of them)
+  - In practice in Fluent we only use a small set of keys
+
+### Cons
+
+- Managing the dependency issues in HVCs and products that have mismatching versions of fluent packages that depend on keyboard-key
```

Comment:
Vnext is in alpha, while this may become temporarily unpleasant experience for our team (need to apologise?... ), I don't think this should be considered as con.